src/main/java/com/google/sps/servlets/EditBusinessDataServlet.java

```diff
@@ -0,0 +1,248 @@
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.gson.Gson;
+import java.lang.reflect.Type;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.Business;
+import static com.google.sps.Constants.*;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Date;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
+import java.text.SimpleDateFormat;  
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns a specific business and lets you edit that specific business */
+@WebServlet("/edit-business-data")
+public class EditBusinessDataServlet extends HttpServlet {
+
+
+  /** Writes a JSON-ified of a specific from the Datastore. The key is taken from the URL parameter.
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+        // Create gson for serializing and deserializing
+        Gson gson = new Gson();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        // TODO: Verify this works once url parameters is sent from previous page
+        Long businessId = Long.parseLong(request.getParameter("businessID"));
```

Comment:
Is this meant to read 
`Long businessId = Long.parseLong(request.getParameter(BUSINESS_ID));`?

Reply:
no, BUSINESS_ID is something i use specifically for my form (it's the name of one of my input elements). I populated that input element with the business id and fetch it again here.

anyway, "businessID" might change depending on Jacob's implementation yeah?

Reply:
Yup that's true. If it's based on your own form, i'm concerned about the confusion here. Can we standardise and use the same const string in our java class? @jacoblipech

Reply:
Let's maybe take this discussion outside the PR?